### PR TITLE
MTSRE-433 | fix: versions to be sorted by Version Parser instead of them being lexicographically sorted

### PR DESCRIPTION
**Pull request by yashvardhan-kukreja**
Signed-off-by: Yashvardhan Kukreja <ykukreja@redhat.com>

### CODE REVIEW

File: managedtenants/bundles/bundle_builder.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 from pathlib import Path
 
 import semver
+from packaging.version import parse as ParseVersion
```
Comment (Ajpantuso) [inline managedtenants/bundles/bundle_builder.py:7]: We already require `semver` so easier to use that. (See [here](https://github.com/mt-sre/managed-tenants-cli/blob/e29e76c215e5c63597f65b54d74220fc3a654287/setup.py#L61)).

You can also just use `max` with the `key` [keyword argument](https://python-semver.readthedocs.io/en/latest/usage/get-min-and-max-of-multiple-versions.html).

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
One comment regarding using `semver`, but the only other missing thing is a regression test. It's probably easiest to just make a helper called `max_version` which given a variadic sequence of strings will return the greatest as parsed by `semver`. Then just take the example from your description and make it the single testcase for a parameterized unit test.

File: managedtenants/bundles/bundle_builder.py (hunk 1 above)
Comment (yashvardhan-kukreja) [reply managedtenants/bundles/bundle_builder.py:7]: Ohh cool, thanks! Will do so :)

**Review by Ajpantuso — APPROVED** [review]
LGTM, we can add a unit test in a following PR.

**Pull request merged**